Upgrade autovalue to 1.7

The automatic update failed. Someone will need to look into this:

> Task :compileJava
warning: [options] bootstrap class path not set in conjunction with -source 1.7
/home/travis/build/googleapis/api-common-java/src/main/java/com/google/api/pathtemplate/PathTemplate.java:34: error: package com.google.auto.value does not exist
import com.google.auto.value.AutoValue;
                            ^
/home/travis/build/googleapis/api-common-java/src/main/java/com/google/api/pathtemplate/PathTemplate.java:171: error: cannot find symbol
  @AutoValue
   ^
  symbol:   class AutoValue
  location: class PathTemplate
2 errors

> Task :compileJava FAILED

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,7 +8,7 @@ addons:
   hostname: fake-hostname-to-work-around-travis-bug
 jdk:
   - oraclejdk8
-  - openjdk7
+  - openjdk8
 script:
   ./gradlew check --info
 after_success:

--- a/build.gradle
+++ b/build.gradle
@@ -139,8 +139,7 @@ configurations {
 
 dependencies {
   codeGeneration libraries.auto_value, libraries.jsr305
-  compile libraries.jsr305
-  compileOnly libraries.auto_value_annotations
+  compile libraries.jsr305, libraries.auto_value_annotations
 }
 
 compileJava.classpath += configurations.codeGeneration

--- a/build.gradle
+++ b/build.gradle
@@ -63,10 +63,10 @@ dependencies {
 
   compile libraries.guava,
     libraries.jsr305,
-    libraries.javax_annotations
-
-  compileOnly libraries.error_prone_annotations,
+    libraries.javax_annotations,
     libraries.auto_value_annotations
+
+  compileOnly libraries.error_prone_annotations
 
   testCompile libraries.junit,
     libraries.mockito,

--- a/build.gradle
+++ b/build.gradle
@@ -39,9 +39,10 @@ ext {
   // Shortcuts for libraries we are using
   libraries = [
       javax_annotations: 'javax.annotation:javax.annotation-api:1.3.2',
+      auto_value_annotations: 'com.google.auto.value:auto-value-annotations:1.7',
+      auto_value: 'com.google.auto.value:auto-value:1.7',
       guava: 'com.google.guava:guava:28.1-android',
       jsr305: 'com.google.code.findbugs:jsr305:3.0.2',
-      autovalue: 'com.google.auto.value:auto-value:1.1',
       error_prone_annotations: 'com.google.errorprone:error_prone_annotations:2.3.3',
 
       // Testing
@@ -57,12 +58,15 @@ repositories {
 }
 
 dependencies {
+
+  annotationProcessor libraries.auto_value
+
   compile libraries.guava,
     libraries.jsr305,
     libraries.javax_annotations
 
-  compileOnly libraries.autovalue,
-    libraries.error_prone_annotations
+  compileOnly libraries.error_prone_annotations,
+    libraries.auto_value_annotations
 
   testCompile libraries.junit,
     libraries.mockito,
@@ -134,9 +138,9 @@ configurations {
 }
 
 dependencies {
-  codeGeneration libraries.autovalue, libraries.jsr305
+  codeGeneration libraries.auto_value, libraries.jsr305
   compile libraries.jsr305
-  compileOnly libraries.autovalue
+  compileOnly libraries.auto_value_annotations
 }
 
 compileJava.classpath += configurations.codeGeneration

